wpimath/src/main/java/edu/wpi/first/math/controller/PIDController.java

```diff
@@ -313,13 +337,18 @@ public boolean isContinuousInputEnabled() {
   }
 
   /**
-   * Sets the minimum and maximum values for the integrator.
+   * Sets the minimum and maximum contributions of the integrator.
+   *
+   * <p>If the integrator value is outside the range [minimumIntegral/ki..maximumIntegral/ki] either
+   * minimumIntegral or maximumIntegral is added to the controller output instead of the integral
+   * gain times the integrator value.
    *
-   * <p>When the cap is reached, the integrator value is added to the controller output rather than
-   * the integrator value times the integral gain.
+   * <p>Additionally, the integrator value is set to minimumIntegral/ki or maximumIntegral/ki
+   * whenever the integrator value would fall outside the range
+   * [minimumIntegral/ki..maximumIntegral/ki].
```

Comment:
This seems like an implementation detail we don't need to document. The prior paragraph is sufficient.

Reply:
This is user-visible, however: future accumulation in the integrator is based on the clamped value, not the out-of-range integrator value. How about
```
/**
  * Sets the minimum and maximum contributions of the integral term.
  *
  * <p>The accumulated integrator value is clamped to the range
  * [minimumIntegral/ki..maximumIntegral/ki]. 
  * Thus, the contribution of the integral term in calculate's result 
  * is always between minimumIntegral and maximumIntegral.
  *
  * @param minimumIntegral The minimum contribution of the integral term
  * @param maximumIntegral The maximum contribution of the integral term
  */
```

Reply:
The user doesn't need to know that's how it's implemented. The only thing they need to care about is the fact that the integral term contribution is clamped between the two values they provided.

Reply:
I'm sorry but I disagree: if the accumulator is allowed to continue growing beyond `maximumIntegral/ki` then when new contributions turn negative they will not immediately reduce the integral term contribution. When the accumulator is clamped they will. It is perhaps a small difference but since there is no separate, detailed, documentation of the integrator range behavior I think it is worth spelling out.

I agree that having to describe the behavior in terms of the quotients with ki is
really not ideal. And maybe how important it is to a person depends on how they think about the i term. I stumbled into this when I was trying to form a mental picture of what physical measurement of my system was being integrated, what units it would have, how the integral would behave based on the physical behavior of the system. Thus my orientation was all about the integral itself. I can see that if one is working with a perspective of "how much can the integral contribution affect my system relative to the proportional and derivative contributions" one might want to ignore the details.

If you still think not including this is adequate, would you find this acceptable:
```
/**
  * Sets the minimum and maximum contributions of the integral term.
  *
  * <p>The contribution of the integral term in calculate's result 
  * is always between minimumIntegral and maximumIntegral.
  *
  * @param minimumIntegral The minimum contribution of the integral term
  * @param maximumIntegral The maximum contribution of the integral term
  */
```

Reply:
You can mention the internal integrator is clamped without saying what it's clamped to, because the user doesn't have to care. For example:
```java
  * <p>The internal integrator is clamped so that the integral term's contribution to the output
  * stays between minimumIntegral and maximumIntegral. This prevents integral windup.
```